src/modules/variables.ts

```diff
@@ -201,7 +204,15 @@ function verifyGivenVariables(
                 missingVars.push(componentExposedVariable);
             }
         } else {
-            if (typeof configuredValue !== componentExposedVariable.type) {
+            if (componentExposedVariable.type === 'array') {
+                if (!Array.isArray(configuredValue)) {
```

Comment:
you could combine the if conditions to avoid that nesting and having duplicate code "wronglyTypedVars.push..."